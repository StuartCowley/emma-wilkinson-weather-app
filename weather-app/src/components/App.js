import React from "react";
import "../styles/App.css";
import LocationDetails from "./LocationDetails";

function App({ location }) {
  console.log('location:', location);

  const { city, country } = location;
  return <LocationDetails city={city} country={country} />;
}

export default App;
