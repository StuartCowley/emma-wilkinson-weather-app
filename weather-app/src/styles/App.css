.weather-app {
    margin: 0 15px;
}
