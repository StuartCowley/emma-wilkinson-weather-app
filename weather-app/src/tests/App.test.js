import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import axios from "axios";
import App from "../components/App";
import forecast from "../data/forecast.json";

describe("App", () => {
  test("renders App component correctly", async () => {
    jest.spyOn(axios, "get").mockResolvedValue({ data: forecast });
    const { asFragment } = render(<App />);
    await waitFor(() => {
      const h1Element = screen.getByText(/Manchester, UK/i);
      expect(h1Element).toBeInTheDocument();
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
