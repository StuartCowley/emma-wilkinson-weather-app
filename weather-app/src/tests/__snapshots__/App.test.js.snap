// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App component correctly 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <h1
      class="location-details"
    >
      Manchester, UK
    </h1>
    <div
      class="search-form"
    >
      <input
        placeholder="Town or city"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Search
      </button>
    </div>
    <div
      class="forecast-summaries"
    >
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary__date"
        >
          Mon Apr 30 2018
        </div>
        <div
          class="forecast-summary__icon"
          data-testid="forecast-icon"
        >
          <img
            alt="weather-icon"
            src="https://cdn-icons-png.flaticon.com/128/382/382573.png"
          />
        </div>
        <div
          class="forecast-summary__temperature"
        >
          11°C
        </div>
        <div
          class="forecast-summary__description"
        >
          Clear
        </div>
        <button
          type="button"
        >
          More details
        </button>
      </div>
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary__date"
        >
          Tue May 01 2018
        </div>
        <div
          class="forecast-summary__icon"
          data-testid="forecast-icon"
        >
          <img
            alt="weather-icon"
            src="https://cdn-icons-png.flaticon.com/128/382/382571.png"
          />
        </div>
        <div
          class="forecast-summary__temperature"
        >
          13°C
        </div>
        <div
          class="forecast-summary__description"
        >
          Stormy
        </div>
        <button
          type="button"
        >
          More details
        </button>
      </div>
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary__date"
        >
          Wed May 02 2018
        </div>
        <div
          class="forecast-summary__icon"
          data-testid="forecast-icon"
        >
          <img
            alt="weather-icon"
            src="https://cdn-icons-png.flaticon.com/128/382/382569.png"
          />
        </div>
        <div
          class="forecast-summary__temperature"
        >
          1°C
        </div>
        <div
          class="forecast-summary__description"
        >
          Heavy Snow
        </div>
        <button
          type="button"
        >
          More details
        </button>
      </div>
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary__date"
        >
          Thu May 03 2018
        </div>
        <div
          class="forecast-summary__icon"
          data-testid="forecast-icon"
        >
          <img
            alt="weather-icon"
            src="https://cdn-icons-png.flaticon.com/128/11103/11103556.png"
          />
        </div>
        <div
          class="forecast-summary__temperature"
        >
          20°C
        </div>
        <div
          class="forecast-summary__description"
        >
          Tornado
        </div>
        <button
          type="button"
        >
          More details
        </button>
      </div>
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary__date"
        >
          Fri May 04 2018
        </div>
        <div
          class="forecast-summary__icon"
          data-testid="forecast-icon"
        >
          <img
            alt="weather-icon"
            src="https://cdn-icons-png.flaticon.com/128/11103/11103556.png"
          />
        </div>
        <div
          class="forecast-summary__temperature"
        >
          25°C
        </div>
        <div
          class="forecast-summary__description"
        >
          Hazy
        </div>
        <button
          type="button"
        >
          More details
        </button>
      </div>
    </div>
    <div
      class="forecast-details"
    >
      <div
        class="forecast-details__date"
      >
        Mon Apr 30 2018
      </div>
      <div
        class="forecast-details__temperature"
      >
        <strong
          class="forecast-details__label"
        >
           Max Temperature: 
        </strong>
        <strong
          class="forecast-details__temperature-max"
        >
          11°C  
        </strong>
        <strong
          class="forecast-details__label"
        >
           Min Temperature: 
        </strong>
        <strong
          class="forecast-details__temperature-min"
        >
          4°C
        </strong>
      </div>
      <div
        class="forecast-details_humidity"
      >
        <strong
          class="forecast-details__label"
        >
           Humidity: 
        </strong>
        <strong
          class="forecast-details__humidity"
        >
           30 % 
        </strong>
      </div>
      <div
        class="forecast-details__wind"
      >
        <strong
          class="forecast-details__label"
        >
           Wind: 
        </strong>
        <strong
          class="forecast-details__wind-direction"
        >
          s
        </strong>
        <strong
          class="forecast-details__wind-speed"
        >
           13 mph 
        </strong>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
